src/toil/cwl/cwltoil.py

```diff
@@ -753,17 +752,40 @@ def size(self, path: str) -> int:
             return super().size(path)
 
 
+def write_to_pipe(file_store, pipeName, id):
+    with file_store.readGlobalFileStream(id) as fi:
+        with open(pipeName, 'wb') as fifo:
+            chunkSz = 1024 * 1024
+            while True:
+                data = fi.read(chunkSz)
+                if not data:
+                    break
+                fifo.write(data)
+
+
 def toil_get_file(
-    file_store: AbstractFileStore, index: dict, existing: dict, file_store_id: str
+    file_store: AbstractFileStore, index: dict, existing: dict, file_store_id: str, streamable: bool
 ) -> str:
     """Get path to input file from Toil jobstore."""
     if not file_store_id.startswith("toilfs:"):
         return file_store.jobStore.getPublicUrl(
             file_store.jobStore.importFile(file_store_id, symlink=True)
         )
-    src_path = file_store.readGlobalFile(FileID.unpack(file_store_id[7:]), symlink=True)
+
+    fileID = FileID.unpack(file_store_id[7:])
+
+    if isinstance(file_store, NonCachingFileStore) and streamable:
```

Comment:
Should skip streaming if the `jobStore` behind this `file_store` is a `fileJobStore`